design/ironic-standalone-networking-service.md

```diff
@@ -0,0 +1,734 @@
+    // +kubebuilder:validation:Maximum=4094
+    NativeVLAN int32 `json:"nativeVlan,omitempty"`
+
+    // AllowedVLANs is a list of VLAN IDs allowed on trunk/hybrid ports
+    // Only valid for trunk and hybrid modes
+    // +optional
+    AllowedVLANs []int32 `json:"allowedVlans,omitempty"`
+}
+
+type BareMetalNetworkAttachmentStatus struct {
+    // Conditions track the state of the network attachment
+    Conditions []metav1.Condition `json:"conditions,omitempty"`
+
+    // AttachedHosts lists the BareMetalHosts currently using this attachment
+    AttachedHosts []string `json:"attachedHosts,omitempty"`
+}
+```
+
+### Enhanced Provision CRD
+
+Add the following fields to the `ProvisionSpec`:
+
+```go
+type ProvisionSpec struct {
+    // ... existing fields ...
+
+    // EnableIronicNetworking enables the Ironic Networking service integration
+    EnableIronicNetworking bool `json:"enableIronicNetworking,omitempty"`
+
+    // ServiceNetwork defines the network configuration for Ironic service operations
+    // This configuration is applied during inspection, cleaning, provisioning, etc.
+    ServiceNetwork *ServiceNetworkConfig `json:"serviceNetwork,omitempty"`
+}
+
+type ServiceNetworkConfig struct {
+    // Mode specifies the switch port mode for service operations
+    Mode PortMode `json:"mode"`
+
+    // NativeVlan specifies the native VLAN ID for service operations
+    NativeVlan int `json:"nativeVlan"`
+
+    // AllowedVlans specifies the list of allowed VLANs for trunk/hybrid modes
+    AllowedVlans []int `json:"allowedVlans,omitempty"`
+}
+```
+
+### Enhanced BareMetalHost CRD
+
+Add the following field to the `BareMetalHostSpec`:
+
+```go
+type BareMetalHostSpec struct {
+    // ... existing fields ...
+
+    // NetworkInterfaces defines the network configuration for each NIC
+    NetworkInterfaces []NetworkInterface `json:"networkInterfaces,omitempty"`
+}
+
+// NetworkInterface defines the network configuration for a specific interface.
+type NetworkInterface struct {
+    // Name of the network interface (e.g., "eth0", "ens1f0")
+    // This must match the name of a NIC in status.hardware.nics.
+    // If MACAddress is set then this field is ignored.
+    // +optional
+    Name string `json:"name"`
+
+    // MAC address of the network interface.  This must match the MAC address
+    // of the network interface in status.hardware.nics.
+    // +optional
+    // +kubebuilder:validation:Pattern=`[0-9a-fA-F]{2}(:[0-9a-fA-F]{2}){5}`
+    MACAddress string `json:"macAddress,omitempty"`
+
+    // NetworkAttachment references the BareMetalNetworkAttachment for this interface
+    // +optional
+    NetworkAttachment *NetworkAttachmentRef `json:"networkAttachment,omitempty"`
+}
+
+type NetworkAttachmentReference struct {
+    // Name of the BareMetalNetworkAttachment
+    Name string `json:"name"`
+
+    // Namespace of the BareMetalNetworkAttachment (defaults to BMH namespace)
+    Namespace string `json:"namespace,omitempty"`
```

Comment:
This may have security implications. We already had to fix a CVE where referencing a secret from another namespace could be used for privilege escalation. A HostNetworkAttachment should not consider anything sensitive.. but my gut feeling is to avoid cross-namespace referencing.

Do you have a strong case for it?